index.js

```diff
@@ -140,6 +140,17 @@ var compile = function(schema, cache, root, reporter, opts) {
         }
       }
     }
+    var errorFromSym = function(sym) {
+      validate('errors++')
+      if (reporter === true) {
+        validate('if (validate.errors === null) validate.errors = []')
+        if (verbose) {
+          validate('validate.errors.push({field:%s,message:%s,value:%s})', JSON.stringify(formatName(name)), sym, name)
+        } else {
+          validate('validate.errors.push({field:%s,message:%s})', JSON.stringify(formatName(name)), sym)
+        }
+      }
+    }
```

Comment:
I don't like this and open to suggestion on how to make `error` and `errorFromSym` to be a single function instead.